portal/src/graphql/portal/DesignScreen/form.ts

```diff
@@ -686,26 +675,17 @@ export function useBrandDesignForm(appID: string): BranchDesignForm {
             return;
           }
           draft.link.color = color;
-          const themeRules = themeRulesStandardCreator();
-          const colorObject = getColorFromString(color);
-          if (colorObject == null) {
+
+          try {
+            const colorShades = getShades(color);
```

Comment:
The getShades function is for v1. So please do not use it.

Please introduce a new function in this file to derive the color. The function should have the following signature

```
function deriveColors(color: string): { original: string, variant: string} | null {
  // The logic present in this file.
  const json = ThemeGenerator.getThemeAsJson(themeRules);
  return {
    original: color,
    variant: json.themeDark,
  };
}
```

Reply:
sure, let me change it

is it also safe to remove the old function? if yes, i will take that away as well to prevent it being used in the future

Reply:
No need to remove it now.

Reply:
You also need to update the tests

Reply:
updated to use derive color function